### DISCUSSION TITLE
is it possible to schedule queries and upload the resultset on remote file server or s3 like storage?

### DISCUSSION THREAD

**Discussion by thadguidry** [9tigerio/db2rest#240] [Ideas]
(banking and telecom friends question)

Ultimately this question brings about the question of if DB2REST should have state management which is needed I think for scheduling purposes.

**Comment by thadguidry** [comment]
This could alternatively be solved by other tools that are good at scheduling and workflow management, like Apache Airflow.  Airflow and its  [HTTP provider operators](https://airflow.apache.org/docs/apache-airflow-providers-http/stable/operators.html#) could be used in an Airflow DAG to schedule a call to DB2REST at some point in time or even scheduled [based on a task updating a dataset](https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/datasets.html#data-aware-scheduling) 

Or [Apache DolphinScheduler ](https://dolphinscheduler.apache.org/en-us) would be my better pick if you had some complex task dependencies or conditions to manage with firing off on a particular schedule or not.

I dunno about this feature request.  Seems already solved with many other tools already robust with timezone support, etc. and that deal with state and calling HTTP and dealing with results, status, conditions.

**Comment by thadguidry** [comment]
When it comes to resultset formats and handling as files.  Dunno, depends on the needs and environments actually.
Good old simple `\o` on a command line can output to a file path.  But you can't get to a command line with DB2REST (thankfully).

I can say that PostgreSQL is very powerful in this regard and even provides things like [PGResult and PQexecParameters](https://www.postgresql.org/docs/current/libpq-exec.html) and the [details of a query result](https://www.postgresql.org/docs/current/libpq-exec.html#LIBPQ-EXEC-SELECT-INFO)

**Comment by susanta-mukhopadhyay** [comment]
It is indeed a good value addition. We may also use the scheduler provided by Spring. We are already using spring boot, it may be easier to us it in the existing framework.
